### PR TITLE
fix(HACBS-1037): fix typos introduced with bindings feature

### PR DESCRIPTION
**Pull request by davidmogar**
Because of the upcoming demo, we had to push some code to main adding support for GitOpsDeployments. This change was not properly reviewed, so this commit address all the problems found.

Signed-off-by: David Moreno García <damoreno@redhat.com>

### CODE REVIEW

**Review by johnbieren — APPROVED** [review]

File: controllers/release/cache.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func SetupReleasePlanAdmissionCache(mgr ctrl.Manager) error {
 		"spec.origin.namespace", releasePlanAdmissionIndexFunc)
 }
 
-// SetupSnapshotEnvironmentBindingCache adds a new index field to be able to search SnapshotEnvironmentBinding by environment.
+// SetupSnapshotEnvironmentBindingCache adds a new index field to be able to search SnapshotEnvironmentBindings by environment.
```
Comment (scoheb) [inline controllers/release/cache.go:48]: ApplicationSnapshotEnvironmentBinding?

Comment (davidmogar) [reply controllers/release/cache.go:48]: The resource is getting renamed (Application prefix is getting dropped) so I renamed it already where I could to reduce the size of the patch once GitOps get it renamed.

**Review by scoheb — COMMENTED** [review]
lgtm

File: controllers/release/cache.go (hunk 1 above)
Comment (scoheb) [reply controllers/release/cache.go:48]: got it. tx

**Pull request merged**